homeassistant/components/nextcloud/sensor.py

```diff
@@ -0,0 +1,102 @@
+"""Summary data from Nextcoud."""
+from datetime import timedelta
+import logging
+
+from nextcloudmonitor import NextcloudMonitor
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_URL,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+# Assign domain
+DOMAIN = "nextcloud"
+# Set Default scan interval in seconds
+SCAN_INTERVAL = timedelta(seconds=60)
+
+# Validate user configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_URL): cv.string,
```

Comment:
Use [cv.url](https://github.com/home-assistant/home-assistant/blob/9e41ee49cbe06bb53bbd5c6d5f3c17a009111b25/homeassistant/helpers/config_validation.py#L552) here:
```suggestion
        vol.Required(CONF_URL): cv.url,
```